src/testing/Testing.hpp

```diff
@@ -82,6 +82,23 @@ boost::test_tools::predicate_result equals(const Eigen::MatrixBase<DerivedA> &A,
   return true;
 }
 
+/// equals to be used in tests. Prints both operatorans on failure
+template <typename TA, typename TB>
```

Comment:
Two remarks: 

- this is only for test purpose, I know, but it would be nice to select more descriptive names for the template arguments. Here, the parameter replace the number type, so maybe `Number` or `NumerType` would be better.
- is it really intended to compare containers with different types? What happens if one container uses `double` precision and the other `float`? Are they still the same if the tolerance is satisfied? According to my understanding, all three input arguments should have the same basic type?

Reply:
> Two remarks:
> 
> * this is only for test purpose, I know, but it would be nice to select more descriptive names for the template arguments. Here, the parameter replace the number type, so maybe `Number` or `NumerType` would be better.

Will do this.

> * is it really intended to compare containers with different types? What happens if one container uses `double` precision and the other `float`? Are they still the same if the tolerance is satisfied? According to my understanding, all three input arguments should have the same basic type?

I will try to change this function to only use a single template parameter.